File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        
        
        def isSymmetric(i, j):
            if not i or not j:
                return i == j
            
            return i.val == j.val and isSymmetric(i.left, j.right) and isSymmetric(i.right, j.left)
        
        return isSymmetric(root, root)
        

File: 0101-symmetric-tree/0101-symmetric-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        
        return isSymmetric(root, root);
        
        
    }
    
    bool isSymmetric(TreeNode *l, TreeNode *r) {
        if (!l || !r) {
            return l == r;
        }
        return l->val == r->val && isSymmetric(l->left, r->right) && isSymmetric(l->right, r->left);
    }
};